### ISSUE TITLE
"Confidence loss is NaN" message is printed

### ISSUE THREAD

**Issue by kimna4** [ljanyst/ssd-tensorflow#24]
Hello,
I know you modified source code through your blog recently.
 So, I tried to use updated source.
But, there is a problem like below when training.
![image](https://user-images.githubusercontent.com/30453641/38850817-0fca536c-424e-11e8-912b-60b0cf00ed3d.png)

[i] Training...
[i] Train  1/200:   3%|▌                  | 28/853 [00:17<08:22,  1.64batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   3%|▋                  | 29/853 [00:17<08:11,  1.68batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   4%|▋                  | 30/853 [00:17<08:00,  1.71batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   4%|▋                  | 31/853 [00:17<07:48,  1.75batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   4%|▋                  | 32/853 [00:17<07:39,  1.79batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   4%|▋                  | 33/853 [00:18<07:30,  1.82batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   4%|▊                  | 34/853 [00:18<07:20,  1.86batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   4%|▊                  | 35/853 [00:18<07:11,  1.90batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   4%|▊                  | 36/853 [00:18<07:03,  1.93batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   4%|▊                  | 37/853 [00:18<06:55,  1.97batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   4%|▊                  | 38/853 [00:18<06:47,  2.00batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   5%|▊                  | 39/853 [00:19<06:40,  2.03batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   5%|▉                  | 40/853 [00:19<06:33,  2.07batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   5%|▉                  | 41/853 [00:19<06:26,  2.10batches/s][!] Confidence loss is NaN.
[i] Train  1/200:   5%|▉                  | 42/853 [00:19<06:20,  2.13batches/s][!] Confidence loss is NaN.

I am using tensorflow 1.8, cuda9-0, cudnn7.

So.. do you know how can I fix it?

Thank you

**Comment by zhuyu72** [comment]
Do you solve the problem? I have the same problem!

**Comment by ljanyst** [comment]
This happens with some versions of tensorflow some of the times. To solve it you need to slightly decrease the initial learning rate.
